vitality/__init__.py

```diff
@@ -471,13 +471,17 @@ def search_workout():
 
         return render_template("workout/search.html")
 
-    @app.route('/workout/<workout_id>', methods=["GET"])
+    @app.route('/workout/<workout_id>', methods=["GET", "POST"])
     def workout(workout_id: str):
         """Page that shows the workout details"""
         if not g.user:
             return redirect(url_for('login'))
 
-        return render_template("workout/workout.html")
+        if (request.method == "POST"):
+            if (g.database.get_workout_class_by_id(workout_id) == None):
+                abort(404)
```

Comment:
I probably said it wrong earlier or something, but /workout route should only be a GET. I think I was thinking of /add_workout when I said it needed a POST request. This page will only need a GET request since it's only returning the workout data.